libraries/CMakeLists.txt

```diff
@@ -1,3 +1,8 @@
 
+include_directories (.)
```

Comment:
Is the effect of this line to let any one of these libraries access the header files of any other one of the libraries?

I see inclusions like:
#include "lib-string-utils/UrlEncode.h"

Therefore lib-telemetry depends on lib-string-utils.  You can also see that from its CMakeLists.txt where it uses target_link_libraries.

But what I'm hoping to do with the libraries I'm developing is avoid #include directives with more than one path component.

And I think target_link_libraries will let you do just that -- shorten the #include path, so long as the library dependent on (here lib-string-utils) does not use PRIVATE in target_sources.

See my commit 45ff9419864c079cbd6f51cdd5e11f58ec882c30.

Reply:
This could be done in a different way, as well as the `*LibName*Api.h`

I can change it if you want, no problem 💯